### PR TITLE
feat: show relevant error messages on the onboarding

### PR DESCRIPTION
**Pull request by MounirDhahri**
The type of this PR is: **Feature**

This PR resolves [CX-1755]

### Description

This PR does two things:
-  Improving the UX by returning user-friendly error messages when the signup fails. The logic is inspired by what we do on the web and that can be found in artsy-passport here https://github.com/artsy/artsy-passport/blob/5f464a55c859d9a6e5f6bccaa92fd3661c88d21d/lib/app/lifecycle.js
- Adding the feature flag to the onboarding welcome screen in order to navigate to the user flow

**Android**

https://user-images.githubusercontent.com/11945712/129567739-4c7ddd41-05d3-4df0-9152-ede226089c49.mov

**iOS**

https://user-images.githubusercontent.com/11945712/129567768-ec5f0ea2-6f8d-49eb-b931-c5fb1900f3ae.mov

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Return friendly user-friendly error messages on failed signups - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- 

</details>

[CX-1755]: https://artsyproduct.atlassian.net/browse/CX-1755

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Return friendly user-friendly error messages on failed signups - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e4663e4a5b41d16c5203975a7a688400e71dc8fd
</p>

File: src/lib/store/AuthModel.ts [hunk 1]
```diff
@@ -12,6 +13,17 @@ import { getCurrentEmissionState } from "./GlobalStore"
 import type { GlobalStoreModel } from "./GlobalStoreModel"
 type BasicHttpMethod = "GET" | "PUT" | "POST" | "DELETE"
 
+const afterSocialAuthLogin = (res: any, reject: (reason?: any) => void, provider: "facebook" | "apple" | "google") => {
+  const providerName = capitalize(provider)
+  if (res.error_description) {
+    if (res.error_description.includes("no account linked to oauth token")) {
```
Comment (MounirDhahri) [inline src/lib/store/AuthModel.ts:19]: Unfortunately this is all we get when trying to sign in.

File: src/lib/store/__tests__/AuthModel-tests.ts [hunk 2]
```diff
@@ -335,7 +335,7 @@ describe("AuthModel", () => {
 
       const result = await GlobalStore.actions.auth.authFacebook({ signInOrUp: "signIn" }).catch((e) => e)
 
-      expect(result).toBe("Could not create user account")
+      expect(result).toBe("Could not log you in")
```
Comment (pvinis) [inline src/lib/store/__tests__/AuthModel-tests.ts:338]: the store action above is `signUp`. shouldnt the error be still `create user account` instead then?

**Review by pvinis — APPROVED** [review]
I'll approve so we can merge and test during QA later today.

Definitely, this feels not ready for release, but let's test this all together in QA rather than just me here.

Assigning you 👉😎👉

File: src/lib/store/__tests__/AuthModel-tests.ts (hunk 2 above)
Comment (MounirDhahri) [reply src/lib/store/__tests__/AuthModel-tests.ts:338]: good point, we were actually returning the wrong error message here. the action we were dispatching was actually a  `signIn` and we were returning a wrong error message.

**Pull request merged**